### ISSUE TITLE
array enums aren't validated correctly

### ISSUE THREAD

**Issue by snrbrnjna** [apigee-127/swagger-tools#343]
Hi there,

i found a related issue in the [swagger-api](https://github.com/swagger-api/swagger-js/issues/198) repo. There are 2 problems:
1. The enums aren't parsed properly. 
2. The enum validation doesn't work for arrays.

My parameter spec:
```yaml
expand:
    name: expand
    description: 'expand inline links to other resources'
    in: query
    type: array
    items:
      type: string
      enum: ['service', 'thing', 'other']
```
The enum isn't parsed this way. it is only if it is specified this way:
```yaml
expandInSeries:
    name: expand
    description: 'expand inline links to other resources'
    in: query
    type: array
    items:
      type: string
    enum: ['service',  'thing', 'other']
```

[`validateEnum()`](https://github.com/apigee-127/swagger-tools/blob/master/lib/validators.js#L251) should validate not only single values but also enums.

**Comment by whitlockjc** [comment]
swagger-js doesn't use swagger-tools and your example is wrong.  You set the `enum` at the top level instead of in the `items` level.  Here is the code that would validate the array items against enums if you set them up properly: https://github.com/apigee-127/swagger-tools/blob/master/lib/validators.js#L618  I'm closing this one but I do think the thing you found in sway might be valid.

**Comment by whitlockjc** [comment]
Whenever you setup an array, it's the `items` schema that is used to validate the array items.  This is not a Swagger or swagger-tools thing but a json-schema thing.

**Comment by snrbrnjna** [comment]
Thanks! I was using an old version (0.9.9). And the array schema validation was fixed meanwhile with dd5195d125e71bdd073dc521f7d3a6f176f39c23.
